frontend/src/lib/components/TaxonomicPopup/TaxonomicPopup.tsx

```diff
@@ -81,3 +84,103 @@ export function TaxonomicPopup({
         </Popup>
     )
 }
+
+/** Like TaxonomicPopup, but convenient when you know you will only use string values */
+export function LemonTaxonomicStringPopup(props: TaxonomicPopupProps<string>): JSX.Element {
+    return (
+        <LemonTaxonomicProps
+            {...props}
+            value={String(props.value)}
+            onChange={(value, groupType) => props.onChange?.(String(value), groupType)}
+            renderValue={(value) => props.renderValue?.(String(value)) ?? <>{String(props.value)}</>}
+        />
+    )
+}
+
+export function LemonTaxonomicProps({
+    groupType,
+    value,
+    onChange,
+    renderValue,
+    groupTypes,
+    dataAttr,
+    eventNames = [],
+    placeholder = 'Please select',
+    style,
+    allowClear = false,
+    ...buttonProps
+}: TaxonomicPopupProps): JSX.Element {
+    const [localValue, setLocalValue] = useState<TaxonomicFilterValue>(value || '')
+    const [visible, setVisible] = useState(false)
+
+    const isClearButtonShown = allowClear && !!localValue
+
+    useEffect(() => {
+        if (!buttonProps.loading) {
+            setLocalValue(value || '')
+        }
+    }, [value])
+
+    return (
+        <div className="LemonButtonWithSideAction">
+            <LemonButtonWithPopup
+                data-attr={dataAttr}
+                popup={{
+                    overlay: (
+                        <TaxonomicFilter
+                            groupType={groupType}
+                            value={value}
+                            onChange={({ type }, payload) => {
+                                onChange?.(payload, type)
+                                setVisible(false)
+                            }}
+                            taxonomicGroupTypes={groupTypes ?? [groupType]}
+                            eventNames={eventNames}
+                        />
+                    ),
+                    sameWidth: false,
+                    actionable: true,
+                    visible,
+                    onClickOutside: () => {
+                        setVisible(false)
+                    },
+                }}
+                onClick={() => {
+                    setVisible(!visible)
+                }}
+                sideIcon={
+                    <div className="side-buttons-row">
+                        {isClearButtonShown && (
+                            <>
+                                <LemonButton
+                                    className="side-buttons-row-button"
+                                    type="tertiary"
+                                    icon={<IconClose fontSize={16} />}
```

Comment:
The `LemonButton` prop `compact` is the canonical way of getting slightly smaller buttons for places like this.

Reply:
Compact is pretty cool but I don't think this is what I'm trying to achieve here. I want to make the icon itself smaller, but compact only shrinks the height of the hoverable area.